app/src/main/java/fr/lille1/univ/android/architecture/btrshop/detailsproduct/DetailsProductFragment.java

```diff
@@ -0,0 +1,194 @@
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Created by charlie on 04/11/16.
+ */
+
+public class DetailsProductFragment extends Fragment implements DetailsProductContract.View {
+
+    //handles the progressbar
+    public ProgressBar mProgressBar;
+
+    private DetailsProductContract.Presenter mPresenter;
+
+    //handles the differents elements
+    @BindView(R.id.products_fragment_brand) TextView brand;
+    @BindView(R.id.products_fragment_category) TextView category;
+    @BindView(R.id.products_fragment_color) TextView color;
+    @BindView(R.id.products_fragment_description) TextView description;
+    @BindView(R.id.products_fragment_dimensions) TextView dimensions;
+    @BindView(R.id.products_fragment_ean) TextView ean;
+    @BindView(R.id.products_fragment_offers) TextView offers;
+    @BindView(R.id.products_fragment_model) TextView model;
+    @BindView(R.id.products_fragment_name) TextView name;
+    @BindView(R.id.products_fragment_weight) TextView weight;
+
+    //handles the differents layout
+    @BindView(R.id.products_fragment_brand_layout) LinearLayout layoutBrand;
+    @BindView(R.id.products_fragment_category_layout) LinearLayout layoutCategory;
+    @BindView(R.id.products_fragment_color_layout) LinearLayout layoutColor;
+    @BindView(R.id.products_fragment_dimensions_layout) LinearLayout layoutDimensions;
+    @BindView(R.id.products_fragment_ean_layout) LinearLayout layoutEan;
+    @BindView(R.id.products_fragment_offers_layout) LinearLayout layoutOffers;
+    @BindView(R.id.products_fragment_model_layout) LinearLayout layoutModel;
+    @BindView(R.id.products_fragment_weight_layout) LinearLayout layoutWeight;
+
+    //handles the cover image
+    @BindView(R.id.products_fragment_logo) ImageView logo;
+    private Bitmap bmp;
+
+
+    public DetailsProductFragment() {
+        // Requires empty public constructor
+    }
+
+    public static DetailsProductFragment newInstance() {
+        return new DetailsProductFragment();
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+    }
+
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View root = inflater.inflate(R.layout.details_product_frag, container, false);
+        ButterKnife.bind(this,root);
+
+        // Retrieve product
+        Product p = (Product) getActivity().getIntent().getSerializableExtra("product");
+        mPresenter.showDetailedProduct(p);
+
+        setHasOptionsMenu(true);
+
+        return root;
+    }
+
+
+    @Override
+    public void showProduct(final Product product) {
+        if (product != null){
+
+            if (product.getBrand() != null){
```

Comment:
You should probably create a method like:

```
private show(TextView view, String value) {
    if (value != null) {
        view.setText(value);
        view.setVisibility(View.VISIBLE);
    }
}
```

And call
```
show(brand, product.getBrand());
show(..., ...);
```
To avoid writing 50 times the same kind of block.